mongo/integration/unified/client_entity.go

```diff
@@ -22,6 +22,10 @@ import (
 	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
+// Security-sensitive commands that should be ignored in command monitoring by default.
+var securitySensitiveCommands = []string{"authenticate", "saslStart", "saslContinue", "getnonce",
+	"createUser", "updateUser", "copydbgetnonce", "copydbsaslstart", "copydb"}
```

Comment:
`hello` and legacy hello commands that contain the `speculativeAuthenticate` field are also sensitive.

This may be very low impact. Per my comment below, handshake commands do not appear in command monitoring events. This would only matter if a test used `RunCommand` to run one of those commands explicitly and needed to omit it from the command started events.

So, if this is non-trivial to check (since it needs to inspect the command body) I would be OK with leaving a TODO comment and addressing this if/when a test like that exists.

Reply:
According to the [security](https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#security) section of the command monitoring spec, we should ignore `hello` and legacy hello commands with `speculativeAuthenticate`. You're right that choosing not to ignore them will make little difference for us, unless there's a test that sends a `hello` and expects it to be ignored (not sure what kind of test that would be).

That said, I've tried to implement ignoring `hello`s (and legacy `hello`s) with `speculativeAuthenticate` in `client_entity#startedEvents()` with a new `isIgnoredEvent` helper. I had to add an `observeSensitiveEvents` to `ClientEntity`, too. Let me know what you think.